igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java

```diff
@@ -428,18 +429,41 @@ public boolean isLogReady(List<Integer> jobIds) {
     return jobIds.stream().map(this::getAndCacheJobLog).allMatch(Optional::isPresent);
   }
 
+  @SuppressWarnings("rawtypes")
   private Optional<String> getAndCacheJobLog(int jobId) {
     log.debug("fetching log by jobId {}", jobId);
     String cachedLog = travisCache.getJobLog(groupKey, jobId);
     if (cachedLog != null) {
       log.debug("Found log for jobId {} in the cache", jobId);
       return Optional.of(cachedLog);
     }
-    V3Log v3Log = travisClient.jobLog(getAccessToken(), jobId);
-    if (v3Log != null && v3Log.isReady()) {
-      log.info("Log for jobId {} was ready, caching it", jobId);
-      travisCache.setJobLog(groupKey, jobId, v3Log.getContent());
-      return Optional.of(v3Log.getContent());
+    try {
+      V3Log v3Log = travisClient.jobLog(getAccessToken(), jobId);
+      if (v3Log != null && v3Log.isReady()) {
+        log.info("Log for jobId {} was ready, caching it", jobId);
+        travisCache.setJobLog(groupKey, jobId, v3Log.getContent());
+        return Optional.of(v3Log.getContent());
+      }
+    } catch (RetrofitError e) {
```

Comment:
I'd of course like to see tests that demonstrate that this error handling (which I like) is working.  I'm not sure if there are good models to follow in igor.  Maybe [ArtifactDownloaderImplTest from rosco](https://github.com/spinnaker/rosco/blob/master/rosco-manifests/src/test/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImplTest.java) is helpful?

I'm slightly hesitant to add more code that depends on RetrofitError.  It's not fair of me to ask for https://github.com/spinnaker/spinnaker/issues/5473 first, though I do want to point it out.  I do happen to know of work going on specifically for igor for that issue, and I suppose this won't make too much more work.

Reply:
Thanks, David! I wasn't aware of that refactor task, but I totally agree that it is something we should do. But maybe not as a part of this PR. But I'll see if I can add a test.

Reply:
Test added 👍